iroha-cli/impl/query_response_handler.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "query_response_handler.hpp"
+
+using namespace iroha::protocol;
+namespace iroha_cli {
+
+  QueryResponseHandler::QueryResponseHandler()
+      : log_(logger::log("QueryResponseHandler")) {}
+
+  void QueryResponseHandler::handle(
+      const iroha::protocol::QueryResponse &response) {
+    if (response.has_error_response()) {
+      handleErrorResponse(response);
+      return;
+    }
+    if (response.has_account_assets_response()) {
```

Comment:
It is better to use map from `iroha::protocol::QueryResponse::ResponseCase` to method